src/Nethermind/Nethermind.Evm/AddressExtensions.cs

```diff
@@ -13,11 +13,13 @@ public static class ContractAddress
     {
         public static Address From(Address? deployingAddress, in UInt256 nonce)
         {
-            ValueKeccak contractAddressKeccak =
-                ValueKeccak.Compute(
-                    Rlp.Encode(
-                        Rlp.Encode(deployingAddress),
-                        Rlp.Encode(nonce)).Bytes);
+            int contentLength = Rlp.LengthOf(deployingAddress) + Rlp.LengthOf(nonce);
```

Comment:
NP: Make this precalculated? Seems like a hot code.

Reply:
I think both of these lengths are constants so probably compiler will replace it, no?

Reply:
only the deployingAddress is constant. Nonce is not constant - its encoded without the leading zeros.
Also, for deployingAddress, there is a null check - so technically not a constant.
```
public static int LengthOf(Address? item)
        {
            return item is null ? 1 : 21;
        }
        ```